Gem/Code/Source/MultiplayerSampleModule.cpp

```diff
@@ -19,6 +19,9 @@
     #include <Components/UI/UiMatchPlayerCoinCountsComponent.h>
     #include <Components/UI/UiRestBetweenRoundsComponent.h>
 #endif
+#if AZ_TRAIT_SERVER  && !AZ_TRAIT_CLIENT
```

Comment:
#if AZ_TRAIT_SERVER && !AZ_TRAIT_CLIENT
Might add comment about this #define. Someone (not me of course *cough*) could get confused and think that a CLIENT could never also be a SERVER and vice-versa. When in reality, both unifiedlauncher.exe and editor.exe are both SERVER and CLIENT (as they both need to handle client-hosted game sessions) and we only want gamelift to run specifically for dedicated servers.